IBM/carbon-addons-iot-react
# fix(build): reconfigure to reference helpers from @babel/runtime

Closes #1073

**Summary**

_I think_ what's happening in the bug report is that by default our async code is being transformed to use `regenerator-runtime` from `node_modules` that pollutes the global scope. This wasn't a problem before we added the `es` build in #1045 because everyone relied on the umd build which included the `regenerator-runtime` in the bundle. It seems some environments may not handle this correctly with the `es` build.  _We are currently_ including this `regenerator-runtime` in `node_modules` of our output build dir:
![image](https://user-images.githubusercontent.com/3360588/78837400-9988a000-79b9-11ea-9470-44af0bd2948f.png)
 I'm unsure why it's not being used in the app environment from the linked bug report.

**No matter the cause, it seems the proper way to handle these helpers is to:**
1. Use helpers from `@babel/runtime` via a hard dep by configuring `@babel/plugin-transform-runtime` so that generator/async functions are transformed to use a `regenerator-runtime` that does not pollute the global scope. [Here's the documentation on this "generator aliasing"](https://babeljs.io/docs/en/babel-plugin-transform-runtime#regenerator-aliasing) and why it's the preferred solution
1. Reconfigure `rollup-plugin-babel` to know not to handle the helpers itself. I added `runtimeHelpers: true,` for this.
1. In addition, I took this as an opportunity to also update babel deps to update to `core-js: 3` so we can take full advantage of `@babel/preset-env`'s `useBuiltIns` option to enable and polyfill instance methods like `"foobar".includes("foo")`. This was not supported with core-js2

**Acceptance Test (how to verify the PR)**

1. create a new folder on your desktop
1. generate a build from `master` branch
1. copy the `es` build directory to the new folder on your desktop, rename it as`./es-old` 
1. generate a build from this branch/PR, and copy the `es` build directory as `./es` to the same folder
1. compare `CardRenderer.jsx` in both builds - this is what you'll see:
![image](https://user-images.githubusercontent.com/3360588/78836471-c340c780-79b7-11ea-87d1-8702fe80b9bb.png)

Helpers should no longer come from `/_virtual/_rollupPluginBabelHelpers.js` but instead come from `@babel/runtime`.
